cli/client.go

```diff
@@ -1197,3 +1209,45 @@ func (c *viamClient) startRobotPartShell(
 	outputLoop()
 	return nil
 }
+
+func parseLogEntryFields(fieldsRaw []*structpb.Struct) ([]zap.Field, error) {
+	parsedFields := make([]zap.Field, 0, len(fieldsRaw))
+	bytes, err := json.Marshal(fieldsRaw)
+	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(bytes, &parsedFields); err != nil {
+		return nil, err
+	}
+	return parsedFields, nil
+}
+
+func logEntryFieldsToString(fields []zap.Field) string {
+	fieldsString := ""
+	for _, field := range fields {
+		fieldValue := ""
+		// TODO use zap encoder to encode
```

Comment:
Would you mind filing a ticket for this and I can eventually use some yet-to-be-created `EncodeZapFields` utility function in the module logging field parsing as well?

Reply:
Deferring till https://github.com/viamrobotics/rdk/pull/3603.

Will swap to that.